sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts

```diff
@@ -75,18 +75,21 @@ export class MSFAANumberSharedService {
       referenceApplicationId,
     );
     const pendingDisbursement =
-      application.currentAssessment.disbursementSchedules.find(
+      application.currentAssessment.disbursementSchedules.filter(
         (schedule) =>
           schedule.disbursementScheduleStatus ===
           DisbursementScheduleStatus.Pending,
       );
-    if (!pendingDisbursement) {
+    if (!pendingDisbursement?.length) {
       throw new CustomNamedError(
         "Not possible to reissue an MSFAA when there is no pending disbursements for the application.",
         APPLICATION_INVALID_DATA_TO_CREATE_MSFAA_ERROR,
       );
     }
-    if (!pendingDisbursement.msfaaNumber.cancelledDate) {
+    const hasPendingAndCancelledMSFAA = pendingDisbursement.some(
```

Comment:
are we using some to check if any of the MSFAA issued is cancelled?

Reply:
Yes, I changed it here after I did a similar check on the UI. It seems better (IMO) in this way, but the idea is still the same: we must have on pending disbursement and this pending disbursement must have a canceled MSFAA.